src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1213,10 +1213,18 @@ def _need_update_container(self):
 
     def set_up_existing_container_update(self):
         if self.get_argument_force_single_container_updates():
+            if len(self.new_containerapp["properties"]["template"]["containers"]) == 0:
+                raise ValidationError(
+                    "The container app must have at least one container. Please use --image to set the image for the container app.")
             # Remove n-1 containers where n is the number of containers in the containerapp and replace remaining container with the new container
             # Fails if all containers are removed
             while (len(self.new_containerapp["properties"]["template"]["containers"]) > 1):
                 self.new_containerapp["properties"]["template"]["containers"].pop()
```

Comment:
Should we keep the container with the same name, not pop them if the container name is eq to `container_name`?

Reply:
No, we should not keep the container with the same name. The goal is to pop everything and only keep the most recently built image. 
So we should keep this logic as-is.